src/quo/components/tags/summary_tag/view.cljs

```diff
@@ -0,0 +1,80 @@
+(ns quo.components.tags.summary-tag.view
+  (:require
+    [clojure.string :as string]
+    [quo.components.avatars.account-avatar.view :as account-avatar]
+    [quo.components.avatars.user-avatar.view :as user-avatar]
+    [quo.components.avatars.wallet-user-avatar.view :as wallet-user-avatar]
+    [quo.components.markdown.text :as text]
+    [quo.components.tags.summary-tag.style :as style]
+    [quo.foundations.colors :as colors]
+    [quo.foundations.resources :as resources]
+    [quo.theme :as quo.theme]
+    [react-native.core :as rn]))
+
+(defn- left-view
+  [{:keys [type token-props account-props collectible-props user-props saved-address-props
+           network-props]}]
+  (case type
+    :account
+    [account-avatar/view
+     {:customization-color (:customization-color account-props)
+      :size                24
+      :emoji               (:emoji account-props)
+      :type                :default}]
+    :network
+    [rn/image
+     {:source (:network-logo network-props)
+      :style  {:width  24
+               :height 24}}]
+    :saved-address
+    [wallet-user-avatar/wallet-user-avatar
+     {:full-name           (:full-name saved-address-props)
+      :size                :size-24
+      :customization-color (:customization-color saved-address-props)}]
+    :collectible
+    [rn/image
+     {:source (:image collectible-props)
+      :style  style/collectible-image}]
+    :user
+    [user-avatar/user-avatar
+     {:full-name           (:full-name user-props)
+      :size                :xs
+      :profile-picture     (:profile-picture user-props)
+      :customization-color (:customization-color user-props)}]
+    :token
+    [rn/image
+     {:source (resources/get-token (keyword (string/lower-case (:symbol token-props))))
+      :style  {:width  24
+               :height 24}}]
+    nil))
```

Comment:
I personally would prefer this as something like:
```
(defn- left-view
  [{:keys [type  emoji customization-color label image-source
           ]}]
  (case type
    :account
    [account-avatar/view
     {:customization-color customization-color
      :size                24
      :emoji               emoji
      :type                :default}]
    :network
    [rn/image
     {:source image-source
      :style  {:width  24
               :height 24}}]
    :saved-address
    [wallet-user-avatar/wallet-user-avatar
     {:full-name           label
      :size                :size-24
      :customization-color customization-color }]
    :collectible
    [rn/image
     {:source image-source
      :style  style/collectible-image}]
    :user
    [user-avatar/user-avatar
     {:full-name           label
      :size                :xs
      :profile-picture     image-source
      :customization-color customization-color}]
    :token
    [rn/image
     {:source image-source
      :style  {:width  24
               :height 24}}]
    nil))
 ```
 and also the `token` image resolving I prefer to do in the application code, as perhaps in some case they are urls and others based on images that we already have in the codebase.
 
 cc @ulisesmac

Reply:
Yep, I agree. The API looks cleaner :+1: